### ISSUE TITLE
OrderedSet subtract returning incorrect result when last remove resizes internal list

### ISSUE THREAD

**Issue by heirendtm** [immutable-js/immutable-js#1603]
It looks like this occurs as of "immutable": "4.0.0-rc.5". Calling subtract on an OrderedSet fails in some cases, and instead returns the original OrderedSet with no items removed. Specifically, if the internal _list happens to be resized on the final remove, it loses the _altered=true flag. Since the internal _list looks unaltered, the subtract function just returns the original OrderedSet.

Try it here: https://codesandbox.io/s/1op9o9475q 

Suspect line is here: https://github.com/facebook/immutable-js/blob/f71ccd5db2016e0dbed72838ab81b931b5e90e9e/src/OrderedMap.js#L139

newList is created containing the correct entries, but on this line the _altered of the newList is false which carries forward to the subsequent checks for if the subtract altered the OrderedSet.  Before that line, list has the altered=true flag. After that, newList has altered=false, and it never gets reset to true since no other items are removed.

**Comment by Anber** [comment]
There is a similar problem with intersect:
```javascript
const r = (f, l) => Array(l - f + 1).fill(0).map((_, i) => f + i);

OrderedSet(r(1, 180)).intersect(OrderedSet(r(1,21))); // returns new set with 21 elements
OrderedSet(r(1, 183)).intersect(OrderedSet(r(1,21))); // returns original set with 183 elements
```

**Comment by jdeniau** [comment]
Thank you for your bug report. The immutable-js oss fork will soon be merged and a fix for this issue has been included in #1833. Once this PR is merged, this issue will be resolved in the main branch. We will then do our best to to release the `4.0.0` version.

Commit reference: https://github.com/immutable-js/immutable-js/pull/1833/commits/7a79603dbdfad44bdf41b4fd73334dce90f9ce64
